src/mlmc/tool/plot.py

```diff
@@ -1042,3 +1457,101 @@ def plot_error(arr, ax, label):
+                             linewidth=self.linewidth,
+                             dashes=[self.linewidth * 1.5] * 2)
+
+
+def violinplot(
+    *,
+    x=None, y=None,
+    hue=None, data=None,
+    order=None, hue_order=None,
+    bw="scott", cut=2, scale="area", scale_hue=True, gridsize=100,
+    width=.8, inner="box", split=False, dodge=True, orient=None,
+    linewidth=None, color=None, palette=None, saturation=.75,
+    ax=None, **kwargs,):
+
+    plotter = ViolinPlotter(x, y, hue, data, order, hue_order,
+                             bw, cut, scale, scale_hue, gridsize,
+                             width, inner, split, dodge, orient, linewidth,
+                             color, palette, saturation)
+
+    if ax is None:
+        ax = plt.gca()
+
+    plotter.plot(ax)
+    return ax
+
+
+def fine_coarse_violinplot(data_frame):
+    fig, axes = plt.subplots(1, 1, figsize=(22, 10))
+
+    # mean with confidence interval
+    # sns.pointplot(x='level', y='samples', hue='type', data=data_frame, estimator=np.mean,
+    #               palette="Set2", join=False, ax=axes)
+
+    # line is not suitable for our purpose
+    # sns.lineplot(x="level", y="samples", hue="type",# err_style="band", ci='sd'
+    #              estimator=np.median, data=data_frame, ax=axes)
+
+    violinplot(x="level", y="samples", hue='type', data=data_frame, palette="Set2",
+                           split=True, scale="area", inner="quartile", ax=axes)
+
+    axes.set_yscale('log')
+    axes.set_ylabel('')
+    axes.set_xlabel('')
+    axes.legend([], [], frameon=False)
+
+    _show_and_save(fig, "violinplot", "violinplot")
+    _show_and_save(fig, None, "violinplot")
+
+
+def plot_pbs_flow_job_time():
+    from mlmc.sample_storage_hdf import SampleStorageHDF
+    import os
+    import mlmc.tool.flow_mc
+    work_dir = "/home/martin/Documents/flow123d_results/flow_experiments/Exponential/corr_length_0_1/sigma_1/L50"
+    sample_storage = SampleStorageHDF(file_path=os.path.join(work_dir, "mlmc_50.hdf5"))
+    sample_storage.chunk_size = 1e8
+    result_format = sample_storage.load_result_format()
+    level_params = sample_storage.get_level_parameters()
+    n_ops = sample_storage.get_n_ops()
+    index = [2, 3, 4, 5, 7, 8]
+    level_params = np.delete(level_params, index)
+    n_ops = np.delete(n_ops, index)
+
+    n_elements = []
+    for level_param in level_params:
+        mesh_file = os.path.join(os.path.join(work_dir, "l_step_{}_common_files".format(level_param)), "mesh.msh")
+        param_dict = mlmc.tool.flow_mc.FlowSim.extract_mesh(mesh_file)
+        n_elements.append(len(param_dict['ele_ids']))
+
+
+    fig = plt.figure(figsize=(15, 8))
+    ax = fig.add_subplot(1, 1, 1)
+    ax.set_xscale('log')
+    lbls = ['{}'.format(nl) for nl in n_elements]
+    ax.set_xticklabels(lbls)
+    #ax.set_yscale('log')
+    ax.plot(1/(level_params**2), n_ops)
+    _show_and_save(fig, "flow_time", "flow_time")
+
+
+if __name__ == "__main__":
```

Comment:
??

Reply:
I suggest making a branch for the preparation of the paper. And then remove (or comment) 
plots that have no general usage.